In [49]:
%%html
<script>
  function code_toggle() {
    if (code_shown){
      $('div.input').hide('500');
      $('#toggleButton').val('Show me Code')
    } else {
      $('div.input').show('500');
      $('#toggleButton').val('Hide Code')
    }
    code_shown = !code_shown
  }

  $( document ).ready(function(){
    code_shown=true;
    $('div.input').hide()
  });
</script>
<form action="javascript:code_toggle()"><input type="submit" id="toggleButton" value="Show Code"></form>

# Capstone project: The Battle of Neighborhoods

## Backgound

I live in Australia thus I want to perform a local case study using Foursquare location data.

Sydney and Melbourne are well-known to be the most famous cities in Australia, and these two cities are both proud of their cultural diversity. However, Sydney and Melbourne are competing with each other in many aspects to show "this city" is the effective cultural capital of Australia.

Hence, in this capstone project, I would like to provide a little evidence that between Sydney and Melbourne, which is the city has a more diverse dining culture.

## Data description

To perform the case study I introduced in the background section, I will choose the central parts of the two cities as the reference locations. Hence, I will choose to use 'Sydney Townhall' as the central point in Sydney, and use 'Melbourne Townhall' as the central point in Melbourne. The exploring radius will be set as 2.5km which is likely to cover the most active areas of the two cities.

The data will be used include:

1. Restaurant name;
2. Restaurant category;
3. Restaurant expense;
4. The number of tips for each restaurant.

Then I will:

1. Use the Foursqure agent to explore the restaurants around the two central points, will a limit of 200;
2. Compare how diverse these restaurants are in terms of styles (say Italian, French, Chinese...);
3. Compare how diverse these restaurants are in terms of expense level (say street food, cheap, medium, expensive, luxury);
4. Compare how popular these restaurants are in terms of the tips in the Foursquare location data.

# Preparation

In [1]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation

#!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize

# !conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library

In [2]:
CLIENT_ID = 'PAZT2ZSYWX2VDQTTXNSKQIWEO4DV2CMFGBHDVGB4JCRRJLRU' # your Foursquare ID
CLIENT_SECRET = 'DCNCLM5SHX33GD4W1QYVMTSOL1RS2AQJT103DYOPSRDRIPTQ' # your Foursquare Secret
VERSION = '20190101'
LIMIT = 200
# print('Your credentails:')
# print('CLIENT_ID: ' + CLIENT_ID)
# print('CLIENT_SECRET:' + CLIENT_SECRET)

# Search for Sydney restaurants

## Get the location of the Sydney Town Hall

In [27]:
address = '483 George Street, Sydney, NSW, AU'
geolocator = Nominatim(user_agent='google')
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

-33.8828509 151.2039662


## Search for results

In [28]:
search_query = 'restaurant'
radius = 2500
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)

In [29]:
results = requests.get(url).json()
# extract venue info and put into df
venues =  results['response']['venues']
SYD_dataframe = json_normalize(venues)
#print(SYD_dataframe.columns)

In [30]:
SYD_dataframe['Cat'] = SYD_dataframe['categories'].apply(lambda x: x[0]['name'])

In [31]:
SYD_dataframe

categories  hasPerk  \
0   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
1   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
2   [{'id': '4bf58dd8d48988d1f5931735', 'name': 'D...    False   
3   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
4   [{'id': '4bf58dd8d48988d113941735', 'name': 'K...    False   
5   [{'id': '4bf58dd8d48988d113941735', 'name': 'K...    False   
6   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
7   [{'id': '52af3b773cf9994f4e043c03', 'name': 'S...    False   
8   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
9   [{'id': '4bf58dd8d48988d1ce941735', 'name': 'S...    False   
10  [{'id': '4bf58dd8d48988d1f5931735', 'name': 'D...    False   
11  [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
12  [{'id': '4bf58dd8d48988d156941735', 'name': 'M...    False   
13  [{'id': '52af3a7c3cf9994f4e043bed', 'name': 'C...    False   
14  [{'id': '4bf58dd8d48988d18f941735', 'name': 'A...    False   
15  [{'id': '4deefc054765f83613cdba6f', 'name': 'I...    False   
16  [{'id': '58daa1558bbb0b01f18ec1cd', 'name': 'L...    False   
17  [{'id': '4bf58dd8d48988d149941735', 'name': 'T...    False   
18  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
19  [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
20  [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
21  [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
22  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
23  [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
24  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
25  [{'id': '58daa1558bbb0b01f18ec1cd', 'name': 'L...    False   
26  [{'id': '4bf58dd8d48988d1ce941735', 'name': 'S...    False   
27  [{'id': '4bf58dd8d48988d14a941735', 'name': 'V...    False   
28  [{'id': '4bf58dd8d48988d142941735', 'name': 'A...    False   
29  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
30  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
31  [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
32  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
33  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
34  [{'id': '4bf58dd8d48988d142941735', 'name': 'A...    False   
35  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
36  [{'id': '4bf58dd8d48988d10b941735', 'name': 'F...    False   
37  [{'id': '4bf58dd8d48988d1c4941735', 'name': 'R...    False   
38  [{'id': '58daa1558bbb0b01f18ec1cd', 'name': 'L...    False   
39  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
40  [{'id': '4bf58dd8d48988d10e941735', 'name': 'G...    False   
41  [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
42  [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
43  [{'id': '4bf58dd8d48988d149941735', 'name': 'T...    False   
44  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
45  [{'id': '4bf58dd8d48988d1ce941735', 'name': 'S...    False   
46  [{'id': '4bf58dd8d48988d1d1941735', 'name': 'N...    False   
47  [{'id': '4bf58dd8d48988d10c941735', 'name': 'F...    False   
48  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...    False   
49  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...    False   

                          id                   location.address location.cc  \
0   4b058766f964a520729022e3                        477 Pitt St          AU   
1   4b058766f964a520939022e3             4-5/683-689 George St.          AU   
2   4d379b19803ba35dc57394fc                        Market City          AU   
3   4b0f2862f964a520ca5f23e3                       41 Dixon St.          AU   
4   4b2323d5f964a520c05324e3                       371A Pitt St          AU   
5   4db7f8d04b22f2ddb6311395                       375 Pitt St.          AU   
6   4b641b59f964a520599f2ae3                      96-100 Hay St          AU   
7   4b73c531f964a52047bb2de3                

Check the restaurants' categories.

In [50]:
#SYD_dataframe['Cat'].unique()

The categories are a bit of messy, thus I set up a dict to organize them by checking both the Cat and Name.

In [33]:
#Set a color dict
SYD_style_dict = {'Chinese Restaurant':'Chinese',
              'Dim Sum Restaurant':'Chinese',
              'Korean Restaurant' : 'Korean',
              'Szechuan Restaurant': 'Chinese',
              'Noodle House':'Chinese',
              'Australian Restaurant':'Australian',
              'Seafood Restaurant': 'Australian',
              'Malay Restaurant':'Malay',
              'Lebanese Restaurant':'Lebanese',
              'Indonesian Restaurant':'Indonesian',
              'Thai Restaurant': 'Thai',
              'Cantonese Restaurant':'Chinese',
              'Art Museum':'Museum',
              'Japanese Restaurant': 'Japanese',
              'Italian Restaurant': 'Italian',
              'Asian Restaurant':'Chinese',
              'Vietnamese Restaurant':'Vietnamese',
              'Restaurant':'Australian',
              'Greek Restaurant':'Greek',
              'Falafel Restaurant':'Falafel',
              'French Restaurant':'French',
              'Bar':'Australian',
              'American Restaurant': 'American'}

In [34]:
SYD_color_dict = {'Chinese':'red','Australian':'blue','Malay':'green','Lebanese':'yellow',
                  'Indonesian':'pink','Thai':'orange','Museum':'black','Japanese':'white',
                  'Italian':'teal','Vietnamese':'grey','Greek':'silver','Falafel':'brown',
                  'French':'plum','Korean':'olive', 'American':'blue'}

## Map the restaurants

In [35]:
Sydney_map = folium.Map(location=[latitude, longitude], zoom_start=14) # generate map centred around the Conrad Hotel

# add a red circle marker to represent the Conrad Hotel
folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='navy',
    tooltip = 'Sydney Town Hall',
    #popup='1 King William Street',
    fill = True,
    fill_color = 'navy',
    fill_opacity = 1
).add_to(Sydney_map)

# add the Italian restaurants as blue circle markers
for lat, lng, label, cat in zip(SYD_dataframe['location.lat'], SYD_dataframe['location.lng'], SYD_dataframe['name'], SYD_dataframe['Cat']):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color=SYD_color_dict[SYD_style_dict[cat]],
        tooltip=SYD_style_dict[cat],
        popup = label,
        fill = True,
        fill_color=SYD_color_dict[SYD_style_dict[cat]],
        fill_opacity=1
    ).add_to(Sydney_map)

# display map
Sydney_map

In [36]:
# Create a new Category column
SYD_dataframe['RCat'] = SYD_dataframe['Cat'].apply(lambda x: SYD_style_dict[x])

## Explore the restaurants to get the price info and number of tips

In [51]:
# SYD_venue_summary = pd.DataFrame(columns=['Name','Category','Price','Tips'])
# for restaurant_id in SYD_dataframe['id']:
#     venue_name = SYD_dataframe[SYD_dataframe['id']==restaurant_id]['name'].values[0]
#     print(venue_name)
#     venue_cat = SYD_dataframe[SYD_dataframe['id']==restaurant_id]['RCat'].values[0]
#     url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(restaurant_id, CLIENT_ID, CLIENT_SECRET, VERSION)
#     results = requests.get(url).json()
#     info =  results['response']['venue']
#     info_df = json_normalize(info)
#     try:
#         price = info_df['price.message'].values[0]
#     except:
#         price = -1
#     tips = info_df['tips.count'].values[0]
#     new_df = pd.DataFrame(columns=['Name','Category','Price','Tips'],
#                               data=[[venue_name,venue_cat,price,tips]])
#     SYD_venue_summary = SYD_venue_summary.append(new_df)

# Search for Melbourne restaurants

## Get the location of the Melbourne Town Hall

In [3]:
address = '90 Swanston St, Melbourne, VIC, AU'
geolocator = Nominatim(user_agent='google')
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

-37.8151593 144.96667


## Search for results

In [4]:
search_query = 'restaurant'
radius = 2500
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)

In [52]:
# results = requests.get(url).json()
# # extract venue info and put into df
# venues =  results['response']['venues']
# MEL_dataframe = json_normalize(venues)
# print(MEL_dataframe.columns)

In [6]:
MEL_dataframe['Cat'] = MEL_dataframe['categories'].apply(lambda x: x[0]['name'])

In [7]:
MEL_dataframe

categories  hasPerk  \
0   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
1   [{'id': '4bf58dd8d48988d1d3941735', 'name': 'V...    False   
2   [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
3   [{'id': '4bf58dd8d48988d1f5931735', 'name': 'D...    False   
4   [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
5   [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
6   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
7   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
8   [{'id': '4bf58dd8d48988d1c4941735', 'name': 'R...    False   
9   [{'id': '4bf58dd8d48988d10f941735', 'name': 'I...    False   
10  [{'id': '4bf58dd8d48988d149941735', 'name': 'T...    False   
11  [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
12  [{'id': '4bf58dd8d48988d142941735', 'name': 'A...    False   
13  [{'id': '4bf58dd8d48988d1d5941735', 'name': 'H...    False   
14  [{'id': '4bf58dd8d48988d113941735', 'name': 'K...    False   
15  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
16  [{'id': '4bf58dd8d48988d142941735', 'name': 'A...    False   
17  [{'id': '4bf58dd8d48988d142941735', 'name': 'A...    False   
18  [{'id': '4bf58dd8d48988d16d941735', 'name': 'C...    False   
19  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
20  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...    False   
21  [{'id': '4bf58dd8d48988d108941735', 'name': 'D...    False   
22  [{'id': '52e81612bcbc57f1066b79f4', 'name': 'B...    False   
23  [{'id': '4bf58dd8d48988d10f941735', 'name': 'I...    False   
24  [{'id': '4bf58dd8d48988d149941735', 'name': 'T...    False   
25  [{'id': '4deefc054765f83613cdba6f', 'name': 'I...    False   
26  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
27  [{'id': '4bf58dd8d48988d10f941735', 'name': 'I...    False   
28  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
29  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...    False   
30  [{'id': '4bf58dd8d48988d113941735', 'name': 'K...    False   
31  [{'id': '4f04af1f2fb6e1c99f3db0bb', 'name': 'T...    False   
32  [{'id': '4bf58dd8d48988d115941735', 'name': 'M...    False   
33  [{'id': '4bf58dd8d48988d10f941735', 'name': 'I...    False   
34  [{'id': '4bf58dd8d48988d10f941735', 'name': 'I...    False   
35  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
36  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
37  [{'id': '4bf58dd8d48988d111941735', 'name': 'J...    False   
38  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
39  [{'id': '4bf58dd8d48988d149941735', 'name': 'T...    False   
40  [{'id': '52af3b773cf9994f4e043c03', 'name': 'S...    False   
41  [{'id': '4bf58dd8d48988d1c4941735', 'name': 'R...    False   
42  [{'id': '4bf58dd8d48988d16d941735', 'name': 'C...    False   
43  [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
44  [{'id': '4bf58dd8d48988d169941735', 'name': 'A...    False   
45  [{'id': '4bf58dd8d48988d10f941735', 'name': 'I...    False   
46  [{'id': '4bf58dd8d48988d113941735', 'name': 'K...    False   
47  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
48  [{'id': '52e81612bcbc57f1066b79ff', 'name': 'H...    False   
49  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...    False   

                          id  \
0   4b05874cf964a520f78922e3   
1   4b05874df964a5204c8a22e3   
2   5036122fe4b0853eeba6e6ec   
3   4b05874df964a520318a22e3   
4   4c10bdba3ce120a1a37c081c   
5   4c261be3db51952140532c3a   
6   4b502c8bf964a520fc1d27e3   
7   4b05874ef964a520c48a22e3   
8   50eb8d1ee4b0ad5f84732696   
9   4b05874ef964a520b28a22e3   
10  4c0ae0bea1b32d7fdb2a9af0   
11  4b05874cf964a520f68922e3   
12  4da81819a86e68edb7d4bc2a   
13  4b726a3df964a520437a2de3   
14  4e77090cd4c0934472942be3   
15  4d1417e2bb64224b6f3daa65   
16  4c5e69ba2815c9289db8b767   
17  5c08984ce4c459002c7e25cb   
18  54dd0cca498e74285a5146a3   
19  4bb9a325935e952166402790   
20  

In [53]:
#MEL_dataframe['Cat'].unique()

In [9]:
#Set a color dict
MEL_style_dict = {'Chinese Restaurant':'Chinese',
              'Dim Sum Restaurant':'Chinese',
              'Korean Restaurant' : 'Korean',
              'Szechuan Restaurant': 'Chinese',
              'Noodle House':'Chinese',
              'Australian Restaurant':'Australian',
              'Seafood Restaurant': 'Australian',
              'Malay Restaurant':'Malay',
              'Lebanese Restaurant':'Lebanese',
              'Indonesian Restaurant':'Indonesian',
              'Thai Restaurant': 'Thai',
              'Cantonese Restaurant':'Chinese',
              'Art Museum':'Museum',
              'Japanese Restaurant': 'Japanese',
              'Italian Restaurant': 'Italian',
              'Asian Restaurant':'Chinese',
              'Vietnamese Restaurant':'Vietnamese',
              'Restaurant':'Australian',
              'Greek Restaurant':'Greek',
              'Falafel Restaurant':'Falafel',
              'French Restaurant':'French',
              'Bar':'Australian',
              'American Restaurant': 'American',
              'Buffet':'Australian',
              'Pub':'Australian',
              'Middle Eastern Restaurant': 'Middle Eastern',
              'Turkish Restaurant':'Middle Eastern',
              'Sushi Restaurant':'Japanese',
              'Dumpling Restaurant':'Chinese',
              'Hotel Bar':'Australian',
              'Vegetarian / Vegan Restaurant':'Japanese',
              'Indian Restaurant':'Indian',
              'Café':'Australian',
              'Halal Restaurant':'Indian'}

In [10]:
MEL_color_dict = {'Chinese':'red','Australian':'blue','Malay':'green','Lebanese':'yellow',
                  'Indonesian':'pink','Thai':'orange','Museum':'black','Japanese':'white',
                  'Italian':'teal','Vietnamese':'grey','Greek':'silver','Falafel':'brown',
                  'French':'plum','Korean':'olive', 'American':'blue','Middle Eastern':'green',
                  'Indian':'orange'}

## Map the restaurants

In [11]:
Melbourne_map = folium.Map(location=[latitude, longitude], zoom_start=14) # generate map centred around the Conrad Hotel

# add a red circle marker to represent the Conrad Hotel
folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='navy',
    tooltip = 'Melbourne Town Hall',
    #popup='1 King William Street',
    fill = True,
    fill_color = 'navy',
    fill_opacity = 1
).add_to(Melbourne_map)

# add the Italian restaurants as blue circle markers
for lat, lng, label, cat in zip(MEL_dataframe['location.lat'], MEL_dataframe['location.lng'], MEL_dataframe['name'], MEL_dataframe['Cat']):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color=MEL_color_dict[MEL_style_dict[cat]],
        tooltip=MEL_style_dict[cat],
        popup = label,
        fill = True,
        fill_color=MEL_color_dict[MEL_style_dict[cat]],
        fill_opacity=1
    ).add_to(Melbourne_map)

# display map
Melbourne_map

## Explore the restaurants to get the price info and number of tips

In [12]:
# Create a new Category column
MEL_dataframe['RCat'] = MEL_dataframe['Cat'].apply(lambda x: MEL_style_dict[x])

In [54]:
# MEL_venue_summary = pd.DataFrame(columns=['Name','Category','Price','Tips'])
# for restaurant_id in MEL_dataframe['id']:
#     venue_name = MEL_dataframe[MEL_dataframe['id']==restaurant_id]['name'].values[0]
#     print(venue_name)
#     venue_cat = MEL_dataframe[MEL_dataframe['id']==restaurant_id]['RCat'].values[0]
#     url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(restaurant_id, CLIENT_ID, CLIENT_SECRET, VERSION)
#     results = requests.get(url).json()
#     info =  results['response']['venue']
#     info_df = json_normalize(info)
#     try:
#         price = info_df['price.message'].values[0]
#     except:
#         price = -1
#     tips = info_df['tips.count'].values[0]
#     new_df = pd.DataFrame(columns=['Name','Category','Price','Tips'],
#                               data=[[venue_name,venue_cat,price,tips]])
#     MEL_venue_summary = MEL_venue_summary.append(new_df)

# Results

In [38]:
import seaborn as sns
import matplotlib.pyplot as plt

In [39]:
%matplotlib inline

   ## Sydney

### Summarize the top 5 categories

In [40]:
SYD_venue_summary.groupby(by='Category').count().sort_values(by='Price',ascending=False).head(5)

Name  Price  Tips
Category                     
Chinese       16     16    16
Australian     8      8     8
Italian        6      6     6
Japanese       5      5     5
Lebanese       3      3     3

### Summarize the price levels 

In [41]:
SYD_venue_summary.groupby(by='Price').count().sort_values(by='Name',ascending=False)

Name  Category  Tips
Price                               
Moderate          25        25    25
Very Expensive    10        10    10
Cheap              8         8     8
-1                 4         4     4
Expensive          3         3     3

### Summarize the tip numbers

Top 10 popular restaurants

In [42]:
SYD_venue_summary.sort_values(by='Tips',ascending=False).head(10)

Name  Category           Price  \
0  Marigold Restaurant 富麗宮 (Marigold Restaurant)   Chinese  Very Expensive   
0                       Sydney Madang Restaurant    Korean        Moderate   
0                          Saké Restaurant & Bar  Japanese  Very Expensive   
0                              Zilver Restaurant   Chinese  Very Expensive   
0              Golden Century Seafood Restaurant   Chinese  Very Expensive   
0                      NaruOne Korean Restaurant    Korean        Moderate   
0            The Eight Modern Chinese Restaurant   Chinese        Moderate   
0             Bill And Toni's Italian Restaurant   Italian        Moderate   
0                       The Restaurant Pendolino   Italian  Very Expensive   
0                   Superbowl Chinese Restaurant   Chinese           Cheap   

  Tips  
0   81  
0   78  
0   77  
0   57  
0   57  
0   47  
0   47  
0   46  
0   41  
0   36

Average number of tips:

In [43]:
SYD_venue_summary['Tips'].mean()

19.76

## Melbourne

### Summarize the top 5 categories

In [45]:
MEL_venue_summary.groupby(by='Category').count().sort_values(by='Price',ascending=False).head(5)

Name  Price  Tips
Category                     
Chinese       15     15    15
Australian    13     13    13
Indian         7      7     7
Italian        3      3     3
Japanese       3      3     3

### Summarize the price levels 

In [46]:
MEL_venue_summary.groupby(by='Price').count().sort_values(by='Name',ascending=False)

Name  Category  Tips
Price                               
Moderate          26        26    26
Cheap             10        10    10
Very Expensive     7         7     7
Expensive          4         4     4
-1                 3         3     3

### Summarize the tip numbers

Top 10 popular restaurants

In [47]:
MEL_venue_summary.sort_values(by='Tips',ascending=False).head(10)

Name        Category           Price Tips
0         Trunk Bar & Restaurant      Australian        Moderate   84
0   Universal Italian Restaurant         Italian        Moderate   50
0           Nam Loong Restaurant         Chinese           Cheap   48
0  Supper Inn Chinese Restaurant         Chinese           Cheap   45
0      Dainty Sichuan Restaurant         Chinese              -1   40
0  Nelayan Indonesian Restaurant      Indonesian        Moderate   38
0         Dragon Boat Restaurant         Chinese        Moderate   36
0           West Lake Restaurant         Chinese  Very Expensive   34
0                Maha Restaurant  Middle Eastern        Moderate   33
0    Kitchen Workshop Restaurant      Australian              -1   25

Average number of tips:

In [48]:
MEL_venue_summary['Tips'].mean()

15.38

# Discussion

1. Restaurant styles

Sydney and Melbourne have different restaurant styles in the city certral area. Although Chinese restaurant holds a biggest proportion in both areas, there are more Australian style restaurants in Melbourne than in Sydney. Moreover, there are more Italian restaurants in the central Sydney while more Indian restaurants in Melbourne.

2. Restaurant price level

In both cities central areas, 'moderate' restaurants account for around 50%. However, there are a little bit more 'very expensive' restaurants and a little bit less 'cheap' restaurants in central Sydney than in central Melbourne. This suggests that the typical meal/dinner expensive may be relatively high in central Sydney.

3. Restaurant popularity measured in the number of tips

While the most commented restaurants in Sydney and Melbourne have similar amount of tips, the average number of Sydney restaurant tips is around 30% higher than that in Melbourne (19.76 vs 15.38). This suggests that if measured by Foursqaure data users, the restaurants in Sydney maybe more popular than those in Melbourne.